pydocx/export/numbering_span.py

```diff
@@ -206,6 +208,103 @@ def get_numbering_level(self, paragraph):
             return None
         return level
 
+    def detect_parent_child_map_for_items(self):
```

Comment:
@jlward @kylegibson would either of you be willing to give this method a look? I understand what this is supposed to do (thanks to the excellent docstring), and I understand the code (looks good), but I don't understand the interaction between numbering_properties, abstract_num_id, and other concepts here to give it a confident review.